### DISCUSSION TITLE
save memory for 128x64 HUB75 matrix

### DISCUSSION THREAD

**Discussion by eadmaster** [MoonModules/WLED-MM#146] [HUB75]
[folowup from here](https://github.com/MoonModules/WLED/pull/108#issuecomment-2227062986)

I've managed to get it working on an `esp32_4MB_V4_S` build by adding this in `bug_manager.cpp` :

````
mxconfig.setPixelColorDepthBits(2);    // must be between 12 and 2. default to 8
````

with this trick all memory allocations went fine and i was able to use the webui to control the matrix.

I've overrided also these (before calling the MatrixPanel_I2S_DMA constructor):
````
      mxconfig.mx_width = 128;
      mxconfig.mx_height = 64;
````

Now everything seems to work as intended, do you have some suggestions to increase the amount of available RAM and possibly increase PixelColorDepthBits to 4?

In my platformio.ini i have defined these:
````
  -D WLED_DISABLE_LOXONE
  -D WLED_DISABLE_ALEXA
  -D WLED_DISABLE_HUESYNC
  -D WLED_DISABLE_MQTT
  -D WLED_DISABLE_INFRARED
  -D WLED_DISABLE_BLYNK
  -D WLED_DISABLE_WEBSOCKETS
````

I've also disabled `USERMOD_AUDIOREACTIVE`:
````
build_unflags = ${common.build_unflags} -D USERMOD_AUDIOREACTIVE
````

**Comment by softhack007** [comment]
> Now everything seems to work as intended, do you have some suggestions to increase the amount of available RAM and possibly increase PixelColorDepthBits to 4?

@eadmaster please try the latest source code from our `mdev` branch. While 128x64 is still an adventure on plain esp32 (and you might get hurt), we have reduced RAM requirements and added automatic color depth reduction based on panel size. You can build with `[env:esp32_4MB_V4_S]` or `[env:esp32_4MB_V4_HUB75_forum]`.
